src/tests/Microsoft.PowerFx.Core.Tests/BindingEngineTests.cs

```diff
@@ -220,6 +220,47 @@ public void CheckParseResultSideEffects()
             var parseResult2 = engine.Parse(formula, options);
             var result2 = engine.Check(parseResult2, options: options);
             Assert.True(result2.IsSuccess);
+        }
+
+        [Fact]
+        public void CheckRecursiveCustomType()
+        {
+            var config = new PowerFxConfig();
+            var engine = new Engine(config);
+
+            var result = engine.Check(
+                "Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop" +
+                ".Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop" +
+                ".Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop" +
+                ".Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop.Loop", new LazyRecursiveRecordType());
+            
+            Assert.True(result.IsSuccess);
+            var recordType = Assert.IsType<RecordType>(result.ReturnType);
+            Assert.Equal("LazyIdentityTest", recordType.Identity.ToString());
+        }
+
+        private class LazyRecursiveRecordType : BaseRecordType
+        {
+            public LazyRecursiveRecordType()
+                : base(new AggregateTypeTests.TestIdentity("LazyIdentityTest"), new List<string> { "Value", "Loop" })

```

Comment:
>AggregateTypeTests.TestIdentity [](http://example.com/codeflow?start=27&length=31)

Idea - can we remove TestIdentity altogether? 
And just use obj.refequals on the derived RecordType as the identity?